pkg/ztunnel/zds/server.go

```diff
@@ -134,7 +134,7 @@ type Server struct {
 
 	restorerPromise promise.Promise[endpointstate.Restorer]
 
-	activeMu sync.Mutex // serialized access to active zc
+	activeMu lock.Mutex // serialized access to active zc
```

Comment:
Why this change?

Reply:
Was recommended by one of our precheck scripts. It would make lock debugging easier.

Reply:
If the `zds` changes are not related to this PR, can we remove them? We will catch any linter errors and fix them on the `zds` focused pull requests.

Reply:
ACK, will remove this.